### PR TITLE
feat(cli): workspace asset registry — evo asset put/get/list/use/rm (closes #55)

### LINKED ISSUE #55: Asset registry: tag and store experiment-produced artifacts for downstream reuse
evo records `artifacts: [{kind, uri, content_key, created_by}]` per experiment result (see `references/glue.md` of the finetuning skill), which is enough to say "exp_0001 produced a LoRA adapter at this path." It does not give the orchestrator a way to:

- **Name an asset and reuse it across experiments.** Today, `exp_0002` referencing `exp_0001`'s checkpoint is done by hardcoding the absolute path into a child experiment's training script. That path is brittle (includes a per-run timestamp + worktree-specific subdir) and silently breaks if the source experiment is gc'd or moved.
- **Query by tag rather than by experiment ID.** "Give me the latest LoRA adapter trained on NuminaMath" -- no way to express. "All datasets tagged as `held-out=true`" -- no way to express.
- **Distinguish produced vs consumed assets.** An experiment that consumed a base model + a curated dataset to produce a checkpoint has no first-class way to record that lineage. The `artifacts` field is producer-only.

This is a generic gap, not a post-training-specific one. Any evo use-case where one experiment's output is another's input -- a built binary used by a downstream perf benchmark, a generated test corpus consumed by a search-algorithm tune, a precomputed embedding index used across retrieval experiments -- hits the same issue.

#### CLI

```
evo asset put <local-path> --name <name> --kind <kind> [--exp <exp_id>] [--tag k=v]...
                                # registers the asset in the workspace's asset registry.
                                # name is the workspace-unique handle ("base-model",
                                # "numina-cot-5k", "exp_0001-adapter").
                                # kind is free-form by default; common: model, dataset,
                                # checkpoint, index, corpus, prompt, config.
                                # --exp ties the asset to one producer experiment;
                                # leave off for workspace-level assets (base model
                                # downloaded once, shared eval corpus, etc.).
                                # --tag adds searchable metadata (--tag epoch=2
                                # --tag verifier=exact-match).

evo asset get <name>            # prints the canonical local path; resolves through
                                # any storage backend the workspace configured
                                # (local, S3, HF Hub, etc.).

evo asset list                  # tabular listing.
evo asset list --kind dataset --tag verifier=exact-match    # filter.
evo asset list --produced-by exp_0001                       # filter.
evo asset list --consumed-by exp_0002                       # filter.

evo asset use <name> --exp <exp_id>
                                # records consumption: exp_0002 reads asset <name>.
                                # Materializes the asset at a stable local path
                                # ($EVO_ASSET_DIR/<name>/...) and emits an env var
                                # ($EVO_ASSET_<NAME_UPPER>=<path>) for the training
                                # script.

evo asset rm <name> [--force]   # remove; refuses if any non-discarded experiment
                                # still references it unless --force.
```

#### Storage model

- Workspace-level registry at `.evo/run_<id>/assets/registry.json` (or top-level `.evo/assets/registry.json` if assets persist across runs).
- Asset files stored under `.evo/run_<id>/assets/<name>/` by default. Backend pluggable -- a `--storage s3://bucket/prefix` or `--storage hf://user/repo` would let large assets live elsewhere and the registry hold the URI.
- The existing per-result `artifacts: []` field continues to work; produced assets registered via `evo asset put --exp <id>` ALSO append to that field automatically, so the two surfaces stay consistent.

#### Env var injection at run time

`evo run` reads each experiment's recorded `consumed_assets[]` (populated by `evo asset use`) and exports `$EVO_ASSET_<NAME>=<resolved-path>` into the benchmark and training subprocess env. The training script reads them by name without ever needing the absolute path:

```python
sft_checkpoint = os.environ.get("EVO_ASSET_EXP_0001_ADAPTER")
if sft_checkpoint:
    model = AutoModelForCausalLM.from_pretrained(sft_checkpoint, ...)
```

This is the same shape as `EVO_PARENT_POLICY` (which already injects the parent experiment's checkpoint path) -- the asset registry generalizes it.

### Why now

The post-training use-case made the gap concrete: 5+ experiments in a single run all needed each other's checkpoints, the only way to share was hardcoding `/home/<user>/.evo/run_0000/worktrees/<parent>/<deep>/<path>/final_model`, and any cleanup of the parent worktree broke every downstream experiment. The same shape will hit any non-trivial evo use-case where experiment outputs are consumed downstream.

### Out of scope for this issue

- Asset content-hashing / deduplication.
- Asset GC policies (keep N latest per kind, etc.).
- Remote storage backends (S3, GCS, HF Hub) -- those are layered on top of a working local registry first.
- UI in the dashboard (Assets tab listing registered assets per experiment) -- nice-to-have, not core.

### PR DESCRIPTION
**Pull request by Srinivasan8888**
## What & why

Closes #55. evo records per-experiment **producer** artifacts, but there's no way to **name** an asset and reuse it across experiments, **query by tag** rather than experiment id, or record **consumed** assets. Downstream experiments hardcode brittle absolute paths that break when a source run is gc'd or moved. This adds a local workspace asset registry.

**Scope:** local-storage-only. Pluggable storage backends (S3 / HF Hub) are intentionally deferred to a follow-up; the schema leaves room for a `backend` field. Kept the surface reviewable rather than shipping the entire proposal at once.

## Surface

```
evo asset put <path> --name N --kind K [--exp E] [--tag k=v]... [--copy]
evo asset get <name>                       # prints canonical local path
evo asset list [--kind] [--tag k=v] [--produced-by] [--consumed-by] [--json]
evo asset use <name> --exp E               # records consumption
evo asset rm <name> [--force]              # refuses while consumed unless --force
```

- `put` registers in place by default; `--copy` materializes under `workspace_path/assets/<name>/`; `--exp` sets `produced_by`. Names are workspace-unique (`rm` first to replace).
- `use` records consumption; from then on the asset path is injected into that experiment's runs as `EVO_ASSET_<NAME>` (e.g. `base-model` → `EVO_ASSET_BASE_MODEL`), so recipes read it by stable handle instead of a hardcoded path.

## Design

- **New module `evo/assets.py`** splits pure registry logic (dict in/out, unit-testable with no workspace) from thin disk wrappers (`load`/`save` under `advisory_lock` + `atomic_write_json`, matching `evo config set`).
- Registry at `workspace_path/assets.json`, **per-run scope** — resets with `evo reset`.
- **Run integration:** `_runtime_env_for_attempt` injects `EVO_ASSET_<NAME>` for every consumed asset, best-effort alongside the existing `EVO_SEED_ARTIFACT` block. Complements `evo new --from-artifact` (per-exp seeding); they coexist.

Registry entry shape:
```json
{ "name": "base-model", "kind": "model", "path": "/abs/path", "tags": {"epoch":"2"},
  "produced_by": "exp_0001", "consumed_by": ["exp_0002"], "copied": false, "created_at": "<utc>" }
```

## Testing

- 16 pure-logic unit tests (put/replace, filter by kind/tags/produced_by/consumed_by, record_use idempotency, remove refuse/force/unknown, env-var mapping, tag parsing) + 10 CLI round-trip tests against a temp workspace.
- Verified end-to-end via the **real CLI** (put in-place + `--copy`, get, list with filters + `--json`, use, rm refuse/force, dup/unknown errors) and a **real `_runtime_env_for_attempt`** call confirming `EVO_ASSET_<NAME>` is injected for a consumer and absent for a non-consumer.
- Full unit suite green (minus pre-existing Rust hook-binary failures unrelated to this change; they reproduce on `main`).

## Out of scope (follow-ups)

Pluggable storage backends (S3/HF), cross-run/global assets (registry is per-run), gc of orphaned materialized copies.

### Note

Issue #55 previously had a full attempt in #80, which its author self-closed (no maintainer objection). This is a fresh, test-driven, deliberately smaller local-first take.

### CODE REVIEW

File: plugins/evo/src/evo/assets.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# --- pure registry logic (no I/O) ------------------------------------------
+
+def registry_put(reg: dict[str, Any], entry: dict[str, Any]) -> dict[str, Any]:
+    """Insert or replace an asset by name. Returns the stored entry."""
+    name = str(entry.get("name") or "").strip()
+    if not name:
+        raise ValueError("asset name must be non-empty")
+    if not str(entry.get("kind") or "").strip():
+        raise ValueError("asset kind must be non-empty")
+    reg.setdefault("assets", {})[name] = entry
```
[the remark this answers is not in the document]
Comment (Srinivasan8888) [reply plugins/evo/src/evo/assets.py]: Fixed in dc0c286. Added `normalize_asset_name()` and applied it at every entry point: `put` normalizes before the dup-check and stores the trimmed name; `registry_put` now rewrites `entry['name']` to match its storage key so the two can't diverge; `get`/`use`/`rm` trim the lookup name; a blank/whitespace-only name is rejected. Verified via the real CLI that `--name ' spaced '` registers under `spaced` and resolves by both the trimmed and padded strings, and that duplicates are caught across padding. New tests cover all three.
